### PR TITLE
[prometheus-pushgateway] Updated push gateway version and ingress to stable

### PR DESCRIPTION
**Pull request by alecrajeev**
#### What this PR does / why we need it:
Updates push gateway to latest version (1.4.1). Also updates ingress to use the stable release for newer versions of kubernets.

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

**Comment by alecrajeev** [comment]
I think there is an older PR related to this, but this also updates ingress. I can merge main branch into his as well as needed.

File: charts/prometheus-pushgateway/templates/ingress.yaml [hunk 1]
```diff
@@ -2,7 +2,9 @@
 {{- $serviceName := include "prometheus-pushgateway.fullname" . }}
 {{- $servicePort := .Values.service.port -}}
 {{- $ingressPath := .Values.ingress.path -}}
-{{- if semverCompare ">=1.14.0-0" .Capabilities.KubeVersion.GitVersion }}
+{{- if semverCompare ">=1.19.0-0" .Capabilities.KubeVersion.GitVersion }}
+apiVersion: networking.k8s.io
```
Comment (monotek) [inline charts/prometheus-pushgateway/templates/ingress.yaml:6]:
```suggestion
apiVersion: networking.k8s.io/v1
```

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/prometheus-pushgateway/templates/ingress.yaml (hunk 1 above)
Comment (alecrajeev) [reply charts/prometheus-pushgateway/templates/ingress.yaml]: Good catch! thanks!

**Review by monotek — DISMISSED** [review]

**Comment by asherf** [comment]
ping @gianrubio , can we get this merged?

**Comment by alecrajeev** [comment]
Apologies I rebased this incorrectly. Created a new PR here: https://github.com/prometheus-community/helm-charts/pull/1306

**Pull request closed without merging**